### PR TITLE
Remove hard requirement to install Rosetta 2 for DD 4.3.0

### PR DESCRIPTION
**Pull request by StefanScherer**
Signed-off-by: Stefan Scherer <stefan.scherer@docker.com>

### Proposed changes

With Docker Desktop 4.3.0 we remove the hard requirement to have Rosetta 2 installed on Apple silicon machines. The Docker Desktop app, the UI, the VM runs natively on Apple silicon. Only a few optional command line tools are still Intel binaries.

### CODE REVIEW

File: desktop/mac/apple-silicon.md [hunk 1]
```diff
@@ -24,21 +24,23 @@ Download Docker Desktop for Mac on Apple silicon:
 
 ### System requirements
 
-You must install **Rosetta 2** as some binaries are still Darwin/AMD64. To install Rosetta 2 manually from the command line, run the following command:
+Beginning with Docker Desktop 4.3.0 we have removed the hard requirement to have **Rosetta 2** installed. There are a few optional command line tools still Darwin/AMD64, see the known issues section below. To have the best experience we suggest to install Rosetta 2. To install Rosetta 2 manually from the command line, run the following command:
```
Comment (stephanierifai) [inline desktop/mac/apple-silicon.md:27]:
```suggestion
Beginning with Docker Desktop 4.3.0 we have removed the hard requirement to have **Rosetta 2** installed. There are a few optional command line tools still when using Darwin/AMD64, see the known issues section below. However, to have the best experience we do suggest installing Rosetta 2. To install Rosetta 2 manually from the command line, run the following command:
```

**Review by stephanierifai — CHANGES_REQUESTED** [review]
LGTM ! One nit

File: desktop/mac/apple-silicon.md (hunk 1 above)
Comment (StefanScherer) [reply desktop/mac/apple-silicon.md:27]: Thanks!

File: desktop/mac/apple-silicon.md [hunk 2]
```diff
@@ -24,21 +24,23 @@ Download Docker Desktop for Mac on Apple silicon:
 
 ### System requirements
 
-You must install **Rosetta 2** as some binaries are still Darwin/AMD64. To install Rosetta 2 manually from the command line, run the following command:
+Beginning with Docker Desktop 4.3.0 we have removed the hard requirement to have **Rosetta 2** installed. There are a few optional command line tools still when using Darwin/AMD64, see the known issues section below. However, to have the best experience we do suggest installing Rosetta 2. To install Rosetta 2 manually from the command line, run the following command:
```
Comment (usha-mandya) [inline desktop/mac/apple-silicon.md:27]:
```suggestion
Beginning with Docker Desktop 4.3.0, we have removed the hard requirement to install **Rosetta 2**. There are a few optional command line tools that still require Rosetta 2 when using Darwin/AMD64. See the Known issues section below. However, to get the best experience, we recommend that you install Rosetta 2. To install Rosetta 2 manually from the command line, run the following command:
```

File: desktop/mac/apple-silicon.md [hunk 3]
```diff
@@ -24,21 +24,23 @@ Download Docker Desktop for Mac on Apple silicon:
...
 
 ```console
 $ softwareupdate --install-rosetta
 ```
 
-We expect to fix this in a future release.
-
 ### Known issues
 
+- Some command line tools are not working when Rosetta 2 is not installed.
```
Comment (usha-mandya) [inline desktop/mac/apple-silicon.md:35]:
```suggestion
- Some command line tools do not working when Rosetta 2 is not installed.
```

File: desktop/mac/apple-silicon.md [hunk 4]
```diff
@@ -24,21 +24,23 @@ Download Docker Desktop for Mac on Apple silicon:
...
+  - The old version 1.x of `docker-compose`. It is recommended to use Compose V2 instead. Either type `docker compose` or enable the `Use Docker Compose V2` in the [General preferences tab](./index.md#general).
```
Comment (usha-mandya) [inline desktop/mac/apple-silicon.md:36]:
```suggestion
  - The old version 1.x of `docker-compose`. We recommend that you use Compose V2 instead. Either type `docker compose` or enable the **Use Docker Compose V2** option in the [General preferences tab](./index.md#general).
```

**Comment by usha-mandya** [comment]
Thanks @StefanScherer. Noted some minor suggestions. Looks good otherwise. We'll also have to update the system requirements on the Install page - https://docs.docker.com/desktop/mac/install/#system-requirements

File: desktop/mac/apple-silicon.md (hunk 2 above)
Comment (StefanScherer) [reply desktop/mac/apple-silicon.md:27]: Thanks!

File: desktop/mac/apple-silicon.md (hunk 3 above)
Comment (StefanScherer) [reply desktop/mac/apple-silicon.md:35]: "do not work", yes, I'll update that.

File: desktop/mac/apple-silicon.md (hunk 4 above)
Comment (StefanScherer) [reply desktop/mac/apple-silicon.md:36]: Thank you.

**Review by stephanierifai — APPROVED** [review]
LGTM !

**Review by usha-mandya — APPROVED** [review]
LGTM. Thanks @StefanScherer

**Pull request merged**